src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,136 @@
+package streams.part1.exercise;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToIntFunction;
+import java.util.stream.Stream;
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getPerson)
+                .map(Person::getAge)
+                .mapToDouble(Double::valueOf)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
+
+        assertEquals(33.66, expected, 0.1);
+    }
+
+    private static int compareForLongestName(Person leftPerson, Person rightPerson) {
+        return rightPerson.getFullName().compareTo(leftPerson.getFullName());
```

Comment:
You should compare names not literally but by length.

Reply:
You are right. Thanks!